include/alpaka/mem/buf/omp5/Set.hpp

```diff
@@ -0,0 +1,116 @@
+/* Copyright 2019 Benjamin Worpitz, Erik Zenker, Matthias Werner
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#ifdef ALPAKA_ACC_ANY_BT_OMP5_ENABLED
+
+#if _OPENMP < 201307
+    #error If ALPAKA_ACC_ANY_BT_OMP5_ENABLED is set, the compiler has to support OpenMP 4.0 or higher!
+#endif
+
+#include <alpaka/dev/DevOmp5.hpp>
+#include <alpaka/kernel/TaskKernelOmp5.hpp>
+#include <alpaka/queue/QueueOmp5Blocking.hpp>
+#include <alpaka/mem/buf/SetKernel.hpp>
+
+#include <alpaka/core/Assert.hpp>
+#include <alpaka/core/Utility.hpp>
+#include <alpaka/dim/DimIntegralConst.hpp>
+#include <alpaka/extent/Traits.hpp>
+#include <alpaka/mem/view/Traits.hpp>
+#include <alpaka/meta/Integral.hpp>
+#include <alpaka/vec/Vec.hpp>
+#include <alpaka/extent/Traits.hpp>
+#include <alpaka/core/Assert.hpp>
+#include <alpaka/idx/Accessors.hpp>
+#include <alpaka/workdiv/WorkDivHelpers.hpp>
+
+#include <cstring>
+
+namespace alpaka
+{
+    namespace dev
+    {
+        class DevOmp5;
+    }
+}
+
+namespace alpaka
+{
+    namespace mem
+    {
+        namespace view
+        {
+            namespace traits
+            {
+                //#############################################################################
+                //! The OMP5 device memory set trait specialization.
+                template<
+                    typename TDim>
+                struct CreateTaskSet<
+                    TDim,
+                    dev::DevOmp5>
+                {
+                    //-----------------------------------------------------------------------------
+                    template<
+                        typename TExtent,
+                        typename TView>
+                    ALPAKA_FN_HOST static auto createTaskSet(
+                        TView & view,
+                        std::uint8_t const & byte,
+                        TExtent const & extent)
+                    {
+                        using Idx = typename idx::traits::IdxType<TExtent>::type;
+                        auto pitch = view::getPitchBytesVec(view);
+                        auto byteExtent = extent::getExtentVec(extent);
+                        byteExtent[TDim::value-1] *= static_cast<Idx>(sizeof(elem::Elem<TView>));
+                        constexpr auto lastDim = TDim::value - 1;
+
+                        if(pitch[0] <= 0)
+                            return kernel::createTaskKernel<acc::AccOmp5<TDim,Idx>>(
+                                    workdiv::WorkDivMembers<TDim, Idx>(
+                                        vec::Vec<TDim, Idx>::ones(),
+                                        vec::Vec<TDim, Idx>::ones(),
+                                        vec::Vec<TDim, Idx>::ones()),
```

Comment:
Isnt't this the "nothing to do"-case? Shouldn't this be zeros?

Reply:
A kernel with a zero workdiv currently leads to a divide-by-zero exception. I did not take zeros to be a valid workdiv.

As it is now, the kernel will run, see that the extent is zero and terminate. With zeros, the `TaskKernelOmp5::operator()` needs to check this and return without executing.

Reply:
I pushed this change: https://github.com/alpaka-group/alpaka/pull/1126/commits/5626aae8e287102b91a52daef51dbe86212eecb0

Reply:
Amended: https://github.com/alpaka-group/alpaka/pull/1126/commits/50666f6941193635a1d1f4f226de3ec5261b7655

Reply:
@psychocoderHPC , @BenjaminW3 : Do you agree with this behavior of `TaskKernelOmp5::operator()`, i.e. return without doing anything and not raising an error if the grid is empty?

Reply:
I am OK with this